### PR TITLE
Fix handling of builtins on Python 2

### LINKED ISSUE #8: Taking qualnames of builtins fails on Py2
E.g.:

```

?  ~[regret:python] -c 'from qualname import qualname; print(qualname(str))'                                                                                          julian@Air ?
Traceback (most recent call last):
  File "<module>", line 1, in <module>
  File "/Users/julian/.local/share/virtualenvs/regret/site-packages/qualname.py", line 50, in qualname
    return obj.__qualname__  # raises a sensible error
AttributeError: type object 'str' has no attribute '__qualname__'
```

whereas `twisted.python.reflect.fullyQualifiedName` (which is what I'm comparing to :) produces the expected:

```

?  ~[regret:python] -c 'from twisted.python.reflect import fullyQualifiedName as qualname; print(qualname(str))'                                                      julian@Air ?
__builtin__.str
```

(other than the fact that qualname isn't fully qualified :()

### PR DESCRIPTION
**Pull request by Julian**
Closes: #8.

### CODE REVIEW

**Comment by Julian** [comment]
Travis appears to have killed 2.6, 3.2 and 3.3

File: qualname.py [hunk 1]
```diff
@@ -47,10 +47,9 @@ def qualname(obj):
     # class/method/function is defined.
     try:
         filename = inspect.getsourcefile(obj)
-    except TypeError:
-        return obj.__qualname__  # raises a sensible error
-    if not filename:
-        return obj.__qualname__  # raises a sensible error
```
Comment (alexmojaki) [inline qualname.py]: I'm pretty sure this part is still important too.

Comment (Julian) [reply qualname.py]: All the tests passed, do you know what case actually hits this branch?

I don't know of when `getsourcefile` returns something falsy, but probably can read through the source to see.

**Comment by Julian** [comment]
OK I added a test here and fixed CI here -- @wbolster not sure if you're still interested in this or not but if so I think ready for review.

**Comment by wbolster** [comment]
tbh, yes i lost interest in this project long ago... it was a fun quick hack that turned out to be useful for others.

but it's only relevant in a mixed python2/3 code base, which i consider a thing of the past, python 2 being end of life, and so on.

anyway, i'll merge this since i don't really have any objections to the proposed changes. thanks for your contribution :+1:

**Review by wbolster — APPROVED** [review]

**Pull request merged**